src/EFCore.Sqlite.Core/Storage/Internal/SqliteRelationalConnection.cs

```
@@ -95,5 +91,19 @@ public virtual ISqliteRelationalConnection CreateReadOnlyConnection()
 
             return new SqliteRelationalConnection(Dependencies.With(contextOptions), _rawSqlCommandBuilder);
         }
+
+        private void InitializeDbConnection(DbConnection connection)
+        {
+            if (_loadSpatialite)
+            {
+                SpatialiteLoader.Load(connection);
+            }
+
+            if (connection is SqliteConnection sqliteConnection
+                && _commandTimeout.HasValue)
+            {
+                sqliteConnection.DefaultTimeout = _commandTimeout.Value;
```

@ajcvickers Is it worth trying to pattern match this for wrapping providers?

---

Thinking about this more:
* If something is functionally required, we'll try pattern matching and warn if we can't.
* If it's not required (like this example), we'll just silently skip it

---

@bricelam Sounds reasonable. I wonder if there is any way we can figure out if wrapping providers are still important in the Core world?